watcher/src/bin/main.rs

```diff
@@ -139,6 +141,7 @@ impl WatcherSyncThread {
         stop_requested: Arc<AtomicBool>,
         logger: Logger,
     ) {
+        let max_blocks_per_iteration = 1000usize;
```

Comment:
This seems like it can be a `const`

Reply:
The way watcher is launched from the binary, it may be better to make this a config option, I can address in an improvement PR subsequent to this.

Reply:
Ah, I didn't realize this is intended to sync to a specific block height.